File: Sorting Algorithms/Bubble Sort/recursive_bubble_sort.py
def recursive_bubble_sort(arr,n):
	for i in range(n-1):
		try:
			if arr[i] > arr[i+1]:
				arr[i],arr[i+1] = arr[i+1],arr[i]
				recursive_bubble_sort(arr,n-1)
		except IndexError:
			pass
	return arr

print('Enter the array:')
arr = input().split(' ')
arr = [int(i) for i in arr]
recursive_bubble_sort(arr,len(arr))
for i in range(len(arr)):
	print(arr[i],end=' ')

File: Sorting Algorithms/Bubble Sort/Recursive Bubble Sort/recursive_bubble_sort.cpp
#include<iostream>

using namespace std;

void Recursive_Bubble_Sort(int arr[], int n)
{
    int i;
    if(n==1)
        return;
    for(i=0;i<n;i++)
    {
       if(arr[i]>arr[i+1])
            swap(arr[i], arr[i+1]);
    }
    Recursive_Bubble_Sort(arr,n-1);
}

void Print(int arr[],int n)
    {
        cout<<"The array after sorting becomes:\n";
        for(int i=0;i<n;i++)
            cout<<arr[i]<<"\t";
    }

int main()
{
    int n;
    cout<<"Enter the size of the array:\n";
    cin>>n;
    int arr[n];
    cout<<"ENter the array elements:\n";
    for(int i=0;i<n;i++)
    {
        cin>>arr[i];
    }
    Recursive_Bubble_Sort(arr,n);
    Print(arr,n);
    return 0;
}
